semgrep_output_v1.atd

```diff
@@ -600,7 +600,7 @@ type cli_match_extra <ocaml attr="deriving show"> = {
    * later on. *)
   ?dataflow_trace: cli_match_dataflow_trace option;
 
-  engine_kind: engine_kind;
+  ?engine_kind: engine_kind option;
```

Comment:
here we need the ?, it's a public interface, it's the official semgrep output.

Reply:
ah I see, I didn't realize that about the `?` (I thought they just went together always)

Reply:
You can read this that I beleive martin wrote a long time ago:
https://atd.readthedocs.io/en/latest/atdgen-tutorial.html?highlight=json#smooth-protocol-upgrades

Reply:
But basically by putting ?engine_kind, we will generate code that will accept the absence of engine_kind,
and so for example in the App, which use code generated from semgrep_output_v1.atd to read the output of semgrep,
they can read an engine_kind if it's there but also accept output where there is no engine_kind (such as with older version of semgrep, that we still want to run sometimes from the playground or also because customer may use older versions).